src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs

```diff
@@ -275,124 +276,148 @@ private object ConvertValue(ExifDataType dataType, ReadOnlySpan<byte> buffer, ui
+            {
+                this.AddInvalidTag(new UnkownExifTag(tag));
+                return;
+            }
 
-            object value;
+            uint size = numberOfComponents * ExifDataTypes.GetSize(dataType);
             if (size > 4)
             {
-                int oldIndex = this.position;
-                uint newIndex = this.ConvertToUInt32(offsetBuffer);
+                uint newIndex = this.ConvertToUInt32(this.offsetBuffer);
 
                 // Ensure that the new index does not overrun the data
-                if (newIndex > int.MaxValue)
+                if (newIndex > int.MaxValue || newIndex + size > this.Length)
                 {
                     this.AddInvalidTag(new UnkownExifTag(tag));
-                    return false;
+                    return;
                 }
 
-                this.position = (int)newIndex;
-
-                if (this.RemainingLength < size)
+                if (this.lazyLoaders.ContainsKey(newIndex))
                 {
-                    this.AddInvalidTag(new UnkownExifTag(tag));
-
-                    this.position = oldIndex;
-                    return false;
+                    Debug.WriteLine($"Duplicate offset: tag={tag}, size={size}, offset={newIndex}");
                 }
 
-                this.TryReadSpan((int)size, out ReadOnlySpan<byte> dataBuffer);
+                this.lazyLoaders.Add(newIndex, () =>
+                {
+                    var dataBuffer = new byte[size];
+                    this.Seek(newIndex);
+                    if (this.TryReadSpan(dataBuffer))
+                    {
+                        object value = this.ConvertValue(dataType, dataBuffer, numberOfComponents);
 
-                value = this.ConvertValue(dataType, dataBuffer, numberOfComponents);
-                this.position = oldIndex;
+                        this.Add(values, exifValue, value);
+                    }
+                });
             }
             else
             {
-                value = this.ConvertValue(dataType, offsetBuffer, numberOfComponents);
-            }
+                object value = this.ConvertValue(dataType, this.offsetBuffer, numberOfComponents);
 
-            exifValue = ExifValues.Create(tag) ?? ExifValues.Create(tag, dataType, numberOfComponents);
+                this.Add(values, exifValue, value);
+            }
+        }
 
-            if (exifValue is null)
+        private void Add(IList<IExifValue> values, IExifValue exif, object value)
+        {
+            if (!exif.TrySetValue(value))
             {
-                this.AddInvalidTag(new UnkownExifTag(tag));
-                return false;
+                return;
             }
 
-            if (!exifValue.TrySetValue(value))
+            foreach (IExifValue val in values)
             {
-                return false;
+                // sometimes duplicates appear,
+                // can compare val.Tag == exif.Tag
+                if (val == exif)
```

Comment:
this is comparing by reference, this seems like a mistake.

Reply:
yeah, this is suspicious, but it used to work like this:
https://github.com/SixLabors/ImageSharp/blob/9f54406aa83002c9e270af937c0b92c1d76d6f83/src/ImageSharp/Metadata/Profiles/Exif/ExifReader.cs#L156